blog/_posts/06-03-blogpost.md

```diff
@@ -0,0 +1,164 @@
+
+---
+layout: post
+title: "BitVector Optimizations and DataMap Features"
+date: 2020-06-03
+author: Jamie Schmidt
+---
+# BitVector Optimizations and DataMap Functionality Improvements
+
+## Introduction 
+For my project, I worked with the Empirical library.  I primarily focused on BitVector and DataMap, two classes for data storage.  
+
+BitVector is, as the name suggests, a Vector of bits (though it's actually bytes).  The standard library has something similar, std::BitMap, but it is incapable of bit operations like NOT or XOR and has a size which must be fixed at compile time.  BitVector, on the other hand, has bit operation support and can be resized during runtime.  However, BitVector did not contain what is known as a "small optimization", that being an optimization where a class has all of its data stored in its own memory, without relying on a pointer elsewhere.   
+
+DataMap is similar to a regular map, but is heavily optimized for scientific computing.  All of its memory is stored in a single array of bytes, and layout information (detailing what is stored where) can be shared between separate DataMaps.  However, DataMap did not contain a way to store a reference to another object in the same place as the object itself.  In other words, given a variable in a DataMap, you couldn't set that variable to be a reference elsewhere.  When you added that variable, you had to choose whether it was going to be the variable itself or a pointer to that variable.  
+
+## BitVector
+
+A BitVector is an eight-byte size and an eight-byte pointer to an array of bytes.  They are generally constructed with a certain length, and all of the bits are set to 0 or 1 based on a boolean you may choose to supply to the constructor.  After that, you can treat it as an array of bits, indexing with Get and Set (or their respective byte and uint variants).  You can also use bit operations such as AND(BitVector bv2) and OR(BitVector bv2), which return the result, or bit operations such as AND_SELF(BitVector bv2) and OR_SELF(BitVector bv2) which return the result and set the BitVector to that result.  If the BitVector needs to become larger or smaller, it can be resized with Resize(size_t newSize).  
```

Comment:
I would also surround the code bits here with tick marks to make them stand out (`Resize(size_t newSize)` instead of Resize(size_t newSize) )